src/validation_workflow/docker/check_docker_daemon.py

```diff
@@ -0,0 +1,27 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+import subprocess
+from typing import Any
+
+"""
+This class is to check if the client (Mac or Linux) has the docker daemon running.
+If the docker daemon is not running, the called shell script will try to bring up the docker daemon.
+It retun 0 if the script runs without error.
+"""
+
+
+class DockerDaemonRunning:
```

Comment:
I dont think we need a different class to check if daemon is running. Add it as one of the validation steps in docker class.

Reply:
Thanks for the comment @gaiksaya .
While testing on my local MAC, I wanted to make sure the docker daemon is running before proceeding further; hende, I added a step to make sure that; and if it is not running, the daemon will be called up.  I am happy to remove this extra step if you would like. thanks!